### PR TITLE
RabbitMQEndpoint code cleanup

### PR DESCRIPTION
**Pull request by FingolfinTEK**
Refactored RabbitMQEndpoint a bit to avoid RabbitMQEndpoint becoming a god class: extracted a few helper classes and did some minor method extractions to improve readability.

### CODE REVIEW

**Comment by davsclaus** [comment]
Do you mind doing a rebase and push as we cannot apply this PR cleanly on master. We do a --rebase for all PRs to ensure they can be applied cleanly on the codebase and we have a nice git tree.

**Comment by FingolfinTEK** [comment]
Sure thing, sorry for troubling you

File: components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQMessageConverter.java [hunk 1]
```diff
@@ -203,18 +202,91 @@ private Object getValidRabbitMQHeaderValue(Object headerValue) {
...
-            return headerValue;
-        } else if (headerValue instanceof Byte) {
-            return headerValue;
-        } else if (headerValue instanceof Double) {
-            return headerValue;
-        } else if (headerValue instanceof Float) {
-            return headerValue;
-        } else if (headerValue instanceof Long) {
-            return headerValue;
         }
```
Comment (FingolfinTEK) [inline components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQMessageConverter.java]: BigDecimal, Byte, Double, Float and Long are all Numbers so no need to explicitly check them too; Timestamp is a java.util.Date so already covered

**Comment by FingolfinTEK** [comment]
Resolved conflicts and pushed again.

**Comment by davsclaus** [comment]
Thanks merging now. 

Btw next time run a checkstyle to fix those also

    mvn compile -Psourcecheck

**Pull request closed without merging**